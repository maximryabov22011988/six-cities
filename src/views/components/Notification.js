import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import cn from 'classnames';
import { get } from 'lodash';

import SvgIcon from './SvgIcon';

import nameSpace from '../../state/name-spaces';

const propTypes = {
  message: PropTypes.string,
  show: PropTypes.bool,
};

class Notification extends React.Component {
  state = {
    show: false,
  };

  componentDidUpdate(prevProps) {
    const { show } = this.props;
    if (prevProps.show !== show && show === true) {
      this.setState({
        show: true,
      });

      this.timer = setTimeout(() => {
        this.setState({
          show: false,
        });
      }, 5000);
    }
  }

  componentWillUnmount() {
    clearTimeout(this.timer);
  }

  handleHide = (evt) => {
    evt.preventDefault();
    this.setState({
      show: false,
    });
  };

  render() {
    const { message } = this.props;
    const { show } = this.state;
    return (
      <div className={cn('notification', show && 'notification--show')}>
        <span>
          <SvgIcon height="20" name="icon-error" width="20" />
          Error
        </span>
        <button className="notification__btn-close" onClick={this.handleHide}>
          <SvgIcon height="14" name="icon-close" width="14" />
        </button>

        {message}
      </div>
    );
  }
}

const mapStateToProps = (state) => ({
  show: Boolean(get(state[nameSpace.APP], 'errors', false)),
  message: get(state[nameSpace.APP], 'errors.message', ''),
});

Notification.propTypes = propTypes;

export default connect(mapStateToProps)(Notification);
