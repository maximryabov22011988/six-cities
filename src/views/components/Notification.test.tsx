import * as React from 'react';
import * as renderer from 'react-test-renderer';

import { Notification } from './Notification';

it('Notification renders correctly with mandatory props', () => {
  const tree = renderer
    .create(
      <Notification show={true} message="Error"/>
    ).toJSON();

  expect(tree).toMatchSnapshot();
});
