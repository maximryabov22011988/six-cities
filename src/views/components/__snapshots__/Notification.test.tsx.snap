// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification renders correctly with mandatory props 1`] = `
<div
  className="notification"
>
  <span>
    <svg
      height="20"
      width="20"
    >
      <use
        xlinkHref="#icon-error"
      />
    </svg>
    <span
      className="visually-hidden"
    />
    Error
  </span>
  <button
    className="notification__btn-close"
    onClick={[Function]}
  >
    <svg
      height="14"
      width="14"
    >
      <use
        xlinkHref="#icon-close"
      />
    </svg>
    <span
      className="visually-hidden"
    />
  </button>
  Error
</div>
`;
